recommender/engine/base_agent_recommender.py

```diff
@@ -38,15 +45,13 @@ def call(self, context: Dict[str, Any]) -> Tuple[int]:
 
         if user is not None:
             return self._for_logged_user(user, search_data, K)
-        else:
-            return self._for_anonymous_user(search_data, K)
+        return self._for_anonymous_user(search_data, K)
 
     def _load_models(self) -> None:
```

Comment:
annotate this class as abstract

Reply:
this is not abstract method (nor class xD)